packages/analytics-browser/test/browser-client.test.ts

```diff
@@ -51,6 +51,48 @@ describe('browser-client', () => {
       expect(parseLegacyCookies).toHaveBeenCalledTimes(1);
     });
 
+    test('should initialize w/o user id and config', async () => {
+      client.setOptOut(true);
+      await client.init(apiKey).promise;
+      expect(client.getUserId()).toBe(undefined);
+    });
+
+    test('should set initalize with undefined user id', async () => {
+      client.setOptOut(true);
+      await client.init(apiKey, undefined).promise;
+      expect(client.getUserId()).toBe(undefined);
+    });
+
+    test('should initialize w/o config', async () => {
+      client.setOptOut(true);
+      await client.init(apiKey, userId).promise;
+      expect(client.getUserId()).toBe(userId);
+    });
+
+    test('should set user id with top level parameter', async () => {
+      client.setOptOut(true);
+      await client.init(apiKey, undefined, {
+        userId,
+      }).promise;
+      expect(client.getUserId()).toBe(undefined);
+    });
+
+    test('should set user to options.userId', async () => {
+      client.setOptOut(true);
+      await client.init(apiKey, {
+        userId,
+      }).promise;
+      expect(client.getUserId()).toBe(userId);
+    });
+
+    test('should set user id using top level parameter as priority', async () => {
```

Comment:
I agree with this but it sort of goes against our standard of `options` overriding previous values (e.g. `track(event, options)`.

That said I think it is probably what most people would expect. So ✅

Reply:
yeah, same thought! this is also status quo for 1.0, so just carried over here